docs/02.API-REFERENCE.md

```diff
@@ -62,6 +62,13 @@ Possible compile time enabled feature types:
  - JERRY_FEATURE_LINE_INFO - line info available
  - JERRY_FEATURE_LOGGING - logging
 
+## jerry_regexp_flags_t
+
+RegExp object optional flags
+  - JERRY_RE_FLAG_GLOBAL - global match; find all matches rather than stopping after the first match
+  - JERRY_RE_FLAG_IGNORE_CASE - ignore case
+  - JERRY_RE_FLAG_MULTILINE - multiline; treat beginning and end characters (^ and $) as working over multiple lines (i.e., match the beginning or end of each line (delimited by \n or \r), not only the very beginning or end of the whole input string)
```

Comment:
In the code, the enumerators already have `JERRY_REGEXP_` prefix. The docs should follow.

Additionally, please, don't let lines run infinitely long. True, the file is not consistent right now. Some paragraphs use 80 chars as limit, some other seem to break lines at 120, yet others seem to be arbitrarily long. Anyhow, do break somewhere as this is too long.